service/controller/clusterapi/v30/cloudconfig/spec.go

```diff
@@ -1,9 +1,13 @@
 package cloudconfig
 
 import (
-	"github.com/aws/aws-sdk-go/service/kms"
+	"context"
+
+	"github.com/giantswarm/certs"
+	"github.com/giantswarm/randomkeys"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 )
 
-type KMSClient interface {
-	Encrypt(*kms.EncryptInput) (*kms.EncryptOutput, error)
+type Interface interface {
+	Render(ctx context.Context, cr v1alpha1.Cluster, clusterCerts certs.Cluster, clusterKeys randomkeys.Cluster, labels string) ([]byte, error)
```

Comment:
This is the interface now. Not pretty, but we have to inject a lot of stuff for the rendering. 🤷‍♂